packages/next-sanity/src/visual-editing/client-component/index.ts

```diff
@@ -114,7 +115,7 @@ export default function VisualEditing(props: VisualEditingProps): null {
     }
 
     return () => disable()
-  }, [basePath, refresh, zIndex])
+  }, [basePath, refresh, rest, zIndex])
```

Comment:
I would rather we continue to pass them explicitly as the amount of options will always be finite, while the `rest` option is using spread syntax, which means it's a new object instance for every render, which will cause the `useEffect` to run tear down + setup on every render 😅

Reply:
Ugh 😬

Reply:
Haha, thanks for also reviewing @stipsan! I'll close this for now, I still like the idea of not having to manually propagate these options everywhere, but the useEffect issue is obviously a big problem (woops/thanks)!